R/html_resources.R

```diff
@@ -269,8 +277,7 @@ discover_rmd_resources <- function(rmd_file, encoding,
               recursive = TRUE,
               include.dirs = FALSE)
             lapply(files, function(f) {
-              list(path = file.path(explicit_res$path, f), 
-                   explicit = TRUE,
+              discover_single_resource(file.path(explicit_res$path, f), TRUE,
```

Comment:
We could probably make this slightly more elegant with some effort but I don't think it'd be a worthwhile investment in this instance; every case is slightly different. 

No, wildcards don't recurse, but naked directories do. It might help to look at `?rmarkdown::find_external_resources`, which describes the globbing behavior in detail. These features are not used very much; I envisioned them as an easy way to stuff all the data needed for a doc into a directory but it seems like for most folks the implicit discovery is doing a good enough job that they don't need this (or perhaps don't know about it). 

Getting back to the code, another possible source of confusion here is that `discover_single_resource` is side effect-y via superassignment. At some point I started calling `discover_single_resource` instead of mutating the resource list directly (to consolidate some sanity checks built in there) but missed a couple cases (and left the dangling variable). This PR corrects those problems, too. 

It's a good idea to add a test case that covers all the behavior here (done).